### PR TITLE
rec: tidy recursor-lua4.cc and recursor-lua4.hh

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

First commit is "mechanical" stuff, 2nd and 3rd more interesting. Especially the elimination of the `goto` should be carefully reviewed. Last commit is reformat only.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
I have no clue about this clan-tidy warning:
```
ext/luawrapper/include/LuaContext.hpp:2891 Potential memory leak (clang-analyzer-cplusplus.NewDeleteLeaks)
```

**Comment by rgacogne** [comment]
> I have no clue about this clang-tidy warning:
> 
> ```
> ext/luawrapper/include/LuaContext.hpp:2891 Potential memory leak (clang-analyzer-cplusplus.NewDeleteLeaks)
> ```

I have seen it before, I'm 99.99% sure it's a false positive but I don't get what confuses clang.

**Comment by Habbie** [comment]
here's how it traces to recursor code https://github.com/PowerDNS/pdns/actions/runs/8230494917/job/22503835322#step:32:769

**Comment by omoerbeek** [comment]
> here's how it traces to recursor code https://github.com/PowerDNS/pdns/actions/runs/8230494917/job/22503835322#step:32:769

In `std_function.h` it calls `new`, but the object has a destructor and it loos like the code cleans up there as well. SO indeed smells like a false positive.

File: pdns/recursordist/lua-recursor4.hh [hunk 1]
```diff
@@ -237,22 +232,45 @@ public:
...
+  }
 
 protected:
   void postPrepareContext() override;
   void postLoad() override;
   void getFeatures(Features& features) override;
 
 private:
+  using gettag_t = std::function<std::tuple<unsigned int, boost::optional<std::unordered_map<int, string>>, boost::optional<LuaContext::LuaObject>, boost::optional<std::string>, boost::optional<std::string>, boost::optional<std::string>, boost::optional<string>> (ComboAddress, Netmask, ComboAddress, DNSName, uint16_t, const EDNSOptionViewMap &, bool, const std::vector<std::pair<int, const ProxyProtocolValue *>> &)>;
+  gettag_t d_gettag; // public so you can query if we have this hooked
```
Comment (chbruyand) [inline pdns/recursordist/lua-recursor4.hh:255]: Not sure the comment is still relevant.

**Review by chbruyand — COMMENTED** [review]
LGTM

**Pull request merged**